tests/sentry/uptime/endpoints/test_validators.py

```diff
@@ -33,3 +37,66 @@ def test(self) -> None:
             )
             == 70
         )
+
+
+class UptimeMonitorDataSourceValidatorTest(TestCase):
+    def get_valid_data(self):
+        return {
+            "url": "https://www.google.com",
+            "interval_seconds": UptimeSubscription.IntervalSeconds.ONE_MINUTE,
+            "timeout_ms": 30000,
+            "method": UptimeSubscription.SupportedHTTPMethods.GET,
+            "headers": [],
+            "trace_sampling": False,
+            "body": None,
+        }
+
+    def setUp(self):
+        self.context = {
+            "organization": self.project.organization,
+            "project": self.project,
+            "request": self.make_request(),
+        }
+
+    def test_simple(self):
+        validator = UptimeMonitorDataSourceValidator(
+            data=self.get_valid_data(), context=self.context
+        )
+        assert validator.is_valid()
+
+    def test_bad_interval(self):
+        data = self.get_valid_data()
+        data["interval_seconds"] = 3700
```

Comment:
nit: probably have `get_valid_data` take a `**kwargs` so you can just inline the overrides